index.js

```diff
@@ -3,67 +3,68 @@ import { migrate } from './lib/migrate.js'
 import getRawBody from 'raw-body'
 import assert from 'http-assert'
 
-export const createHandler = async (client) => {
-  await migrate(client)
+const handler = async (req, res, client) => {
+  const segs = req.url.split('/').filter(Boolean)
+  if (segs[0] === 'retrievals' && req.method === 'POST') {
+    // TODO: Consolidate to one query
+    const { rows: [retrievalTemplate] } = await client.query(`
+      SELECT id, cid, provider_address, protocol
+      FROM retrieval_templates
+      OFFSET floor(random() * (SELECT COUNT(*) FROM retrieval_templates))
+      LIMIT 1
+    `)
+    const { rows: [retrieval] } = await client.query(`
+      INSERT INTO retrievals (retrieval_template_id)
+      VALUES ($1)
+      RETURNING id
+    `, [
+      retrievalTemplate.id
+    ])
+    json(res, {
+      id: retrieval.id,
+      cid: retrievalTemplate.cid,
+      providerAddress: retrievalTemplate.provider_address,
+      protocol: retrievalTemplate.protocol
+    })
+  } else if (segs[0] === 'retrievals' && req.method === 'PATCH') {
+    const retrievalId = Number(segs[1])
+    assert(!Number.isNaN(retrievalId), 400, 'Invalid Retrieval ID')
+    const body = await getRawBody(req, { limit: '100kb' })
+    const { success } = JSON.parse(body)
+    const { rows } = await client.query(`
+      UPDATE retrievals
+      SET finished_at = NOW(),
+          success = $1
+      WHERE id = $2
+      RETURNING id
+    `, [
+      success,
+      retrievalId
+    ])
+    assert(rows.length > 0, 404, 'Retrieval Not Found')
+    res.end('OK')
+  } else {
+    res.end('Hello World!')
+  }
+}
```

Comment:
While you are refactoring this file, would you mind extracting route handlers into individual functions?

```js
const handler = async (req, res, client) => {
const segs = req.url.split('/').filter(Boolean)
if (segs[0] === 'retrievals' && req.method === 'POST') {
  createNewRetrieval(req, res, client);
} else if (segs[0] === 'retrievals' && req.method === 'PATCH') {
  patchRetrievalById(Number(segs[1]), req, res, client);
} else { 
  res.end('Hello World!');
}
```